src/webgpu/util/device_pool.ts

```diff
@@ -234,42 +237,55 @@ function supportsFeature(
 type DeviceHolderState = 'free' | 'reserved' | 'acquired';
 
 /**
- * Holds a GPUDevice and tracks its state (free/reserved/acquired) and handles device loss.
+ * Holds GPUDevice with same descriptor and tracks device state (free/reserved/acquired) and handles device loss.
+ * All the devices must be free/reserved/acquired together.
```

Comment:
The idea of getting multiple devices from the device pool was to increase the pool's size so it could have more than 1 default device holder, rather than having multiple devices in each device holder. The device holder is designed to hold just one device (e.g. it has only one lostInfo). Extending it to hold 2 works for these tests, but seems overspecialized for this exact use case, I don't think it can be used for much else.

This new code _is_ overall nicer even though it's weird in a different way, but it also makes *every test* use 2 devices. That's probably not super costly, but it is surprising and I'd prefer not to.

Possible to split new test coverage and these framework changes? I'd like to land the new tests but I'm wary of the 2-device thing which could have effects on all the tests.

Reply:
OK, I will split the new tests to a separate PR, we can land it first. About getting multiple devices, I have different thought, let's dicuss it in another PR when you get back.